### ISSUE TITLE
Convert quest_id from long id to char* and refactor functions accordingly

### ISSUE THREAD

**Issue by gabigarc03** [uchicago-cs/chiventure#1158]
Quests currently use a long int to store IDs, which can be confusing. String identifiers would be much more user-friendly, hence this change. See the [current implementation wiki](https://github.com/uchicago-cs/chiventure/wiki/Quests-~-Current-Implementation) for more details.

**Comment by Castel-Jaloux** [comment]
I have made the changes and pushed to a separate branch of the project. I debugged all the errors that arose. I have not yet submitted a pull request for this change.

**Comment by Castel-Jaloux** [comment]
In this task, I changed the type of quest_id from long int to char* for improved usability. In addition to changing the code, I altered examples, implementation of functions, and testing code in order to ensure that the changes were properly accounted for in these aspects. Additionally I ran through testing several times to remove bugs that cropped up as a result of the change. I then submitted a pull request, which is pending approval. Further comments on this issue will be documented in the linked pull requests's comment chain.
